src/EditorFeatures/VisualBasic/LineCommit/CommitBufferManager.vb

```diff
@@ -215,8 +235,20 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.LineCommit
                 ' We want to include the line break into the line before
                 finalSpanStart = dirtySpan.Snapshot.GetLineFromLineNumber(startingLine.LineNumber - 1).End
             End If
-
-            formattingInfo.SpanToFormat = New SnapshotSpan(dirtySpan.Snapshot, Span.FromBounds(finalSpanStart, finalSpanEnd))
+            If formattingInfo._useStatementSemanticFormatting Then
+                Dim hyridStartingLine = dirtySpan.Snapshot.GetLineFromPosition(semanticSpanStart)
+                If hyridStartingLine.LineNumber = 0 Then
+                    semanticSpanStart = 0
+                Else
+                    ' We want to include the line break into the line before
+                    semanticSpanStart = dirtySpan.Snapshot.GetLineFromLineNumber(hyridStartingLine.LineNumber - 1).End
+                End If
+                formattingInfo._statementSpanToSemanticFormat = New SnapshotSpan(dirtySpan.Snapshot, Span.FromBounds(semanticSpanStart, semanticSpanEnd))
+            End If
+            If semanticSpanStart = finalSpanStart AndAlso semanticSpanEnd = finalSpanEnd Then
+                formattingInfo._useStatementSemanticFormatting = False
+            End If
+            formattingInfo._spanToFormat = New SnapshotSpan(dirtySpan.Snapshot, Span.FromBounds(finalSpanStart, finalSpanEnd))
             Return True
```

Comment:
this all needs comments explaining what is going on .  i don't understand the reasoning for any of it.

Reply:
Done